src/context.cpp

```diff
@@ -281,13 +303,13 @@ namespace librealsense
 #endif //BUILD_WITH_DDS
 
         // Supported recovery devices
-        if (mask & RS2_PRODUCT_LINE_D400) 
+        if( mask & RS2_PRODUCT_LINE_D400 && ! ( mask & RS2_PRODUCT_LINE_SW_ONLY ) )
         {
             auto recovery_devices = fw_update_info::pick_recovery_devices(ctx, devices.usb_devices, mask);
             std::copy(begin(recovery_devices), end(recovery_devices), std::back_inserter(list));
         }
 
-        if( mask & RS2_PRODUCT_LINE_NON_INTEL )
+        if( mask & RS2_PRODUCT_LINE_NON_INTEL && ! ( mask & RS2_PRODUCT_LINE_SW_ONLY ) )
```

Comment:
Note that this kind of changes needs to be added on other products as well and will not shown as conflicts.

We need to be carefull with those refactors

Reply:
This is compatible with all products. The flag is the `--sw-only` flag is is mostly on for our development and in unit-tests. We are careful.

Reply:
I'll add the flag when I do the PR with the 555e